File: exercises/java/src/main/java/cards/MixedDeck.java
package cards;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class MixedDeck implements Deck {

    private List<Card> cards = new ArrayList<>();

    public MixedDeck() {
        Deck animalDeck = new AnimalDeck();
        Deck playingCardDeck = new PlayingCardDeck();
        for(int cardNo=0; cardNo<52; cardNo++) {
            cards.add(animalDeck.deal());
            cards.add(playingCardDeck.deal());
        }
    }

    @Override
    public void shuffle() {
        Collections.shuffle(cards);
    }

    @Override
    public String[] getCards() {
        String[] result = new String[cards.size()];
        for(int cardIndex=0; cardIndex<cards.size(); cardIndex++){
            result[cardIndex] = cards.get(0).toString();
        }
        return result;
    }

    @Override
    public Card deal() {
        return cards.remove(0);
    }
}

File: exercises/php/cards/MixedDeck.php
<?php

namespace McrDigital\PhpFundamentals1\Cards;

class MixedDeck implements Deck
{
    private array $cards;

    public function __construct()
    {
        $this->cards = [];

        $animalDeck = new AnimalDeck();
        $playingCardDeck = new PlayingCardDeck();

        for ($cardNo = 0; $cardNo < 52; $cardNo++) {
            $this->cards[] = $animalDeck->deal();
            $this->cards[] = $playingCardDeck->deal();
        }
    }

    public function getCards(): array
    {
        $result = [];
        foreach ($this->cards as $card) {
            $result[] = strval($card);
        }
        return $result;
    }

    public function deal(): Card
    {
        $card = array_splice($this->cards, 0, 1)[0];
        return $card;
    }

    public function shuffle(): void
    {
        for ($i = 0; $i < count($this->cards); ++$i) {
            $indexA = rand(0, $i);
            $indexB = $i;

            $valueA = $this->cards[$indexA];
            $valueB = $this->cards[$indexB];

            $this->cards[$indexA] = $valueB;
            $this->cards[$indexB] = $valueA;
        }
    }
}